### PR TITLE
feat(git-submodules): default cloneSubmodules to false

### PR DESCRIPTION
**Pull request by rarkins**
## Changes:

Defaults `cloneSubmodules` to `false`.

## Context:

BREAKING CHANGE: Git Submodules cloning now needs to be explicitly configured.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by JamieMagee — DISMISSED** [review]
Makes sense. This should stop any new issues being created until I can make my way through the backlog. Thanks!

File: docs/usage/configuration-options.md [hunk 1]
```diff
@@ -285,6 +285,10 @@ Make sure that version is a pinned version of course, as otherwise it won't be v
 
 ## cloneSubmodules
 
+Enabling this option will mean that any detected git submodules will be cloned at time of repository clone.
```
Comment (HonkingGoose) [inline docs/usage/configuration-options.md:288]:
```suggestion
Enabling this option will mean that any detected Git submodules will be cloned at time of repository clone.
```

Maybe we should capitalize the proper noun Git?

File: docs/usage/configuration-options.md [hunk 2]
```diff
@@ -285,6 +285,10 @@ Make sure that version is a pinned version of course, as otherwise it won't be v
...
+
+Important: private submodules aren't yet supported by Renovate, unless the underlying `ssh` layer already has the appropriate permissions.
```
Comment (HonkingGoose) [inline docs/usage/configuration-options.md:290]:
```suggestion
Important: private submodules aren't supported by Renovate, unless the underlying `ssh` layer already has the appropriate permissions.
```

**Review by HonkingGoose — CHANGES_REQUESTED** [review]
Some small niggles.

Also I'm missing an explanation on **why** enabling cloneSubmodules is/can be better than the default behavior.

Also I'd like an explanation on how to know/see if the underlying SSH layer has appropriate permissions.

**Pull request merged**